exercises/saddle-points/.meta/template.j2

```diff
@@ -0,0 +1,58 @@
+"""Tests for the saddle-points exercise
+
+Implementation note:
+The saddle_points function must validate the input matrix and raise a
+ValueError with a meaningful error message if the matrix turns out to be
+irregular.
+"""
+{%- import "generator_macros.j2" as macros with context -%}
+
+{% macro test_case(case) -%}
+    {%- set input = case["input"] -%}
+    {%- set expected = case["expected"] -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        {% if input["matrix"][0] -%}
+        matrix = {{ input["matrix"] }}
+        {%- else -%} matrix = [] {% endif %}
+        {% if expected|length > 2 -%}
+        expected = {{ list_points(expected) }}
+        self.assertEqual({{ case["property"] | to_snake }}(matrix), expected)
+        {% else -%}
+        self.assertEqual(
+            {{- case["property"] | to_snake }}(matrix), {{ list_points(expected) }})
+        {%- endif %}
+{%- endmacro -%}
+
+{% macro list_points(points) -%}
+    {% if points -%}
+    {{ points | sort(attribute="column") }}
+    {% else -%}
+    [{}]
```

Comment:
I am not sure why the current implementation prefers `[{}]` over `[]` for an "empty" result; I am in favor of changing this (requires change to example.py). This would eliminate the need for this macro altogether.

@yawpitch, are you good with changing the expectation for empty results?

Reply:
@cmccandless yeah, I've no idea why that was ever done. The canonical data seems to expect just an empty list anyway, so eliminating the disconnect seems sensible.

Reply:
Yeah, I thought that was a little weird. I've made the change.

I also switched from using `assertEqual` to `assertCountEqual` which will handle the possibility of the points being in a different order in the list and remove the need to sort.

Reply:
Looks like `assertCountEqual` is not in Python 2.7. Reverted to sorting, but I added a more robust sorting logic than previously.